### PR TITLE
V3 parity spec for enduser email consent view

### PR DESCRIPTION
**Pull request by vijetmahabaleshwar-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-594859](https://oktainc.atlassian.net/browse/OKTA-594859)

### CODE REVIEW

File: test/testcafe/framework/page-objects/EnduserConsentPageObject.js [hunk 1]
```diff
@@ -13,15 +14,31 @@ export default class EnduserConsentPageObject extends ConsentPageObject {
...
+      return this.form.getElement('[data-type="save"]').innerText; 
+    }
     return this.form.getElement('input[type="submit"]').value;
   }
 
   getCancelButtonLabel() {
+    if (userVariables.v3) {
+      return this.form.getElement('[data-type="cancel"]').innerText; 
+    }
     return this.form.getElement('input[data-type="cancel"]').value;
```
Comment (shawyu-okta) [inline test/testcafe/framework/page-objects/EnduserConsentPageObject.js:42]: Can you use `this.form.getButton('Cancel')` and then divide the behavior on `innerText` vs `value` if necessary after that? This applies to `getSaveButtonLabel` too

Comment (shawyu-okta) [reply test/testcafe/framework/page-objects/EnduserConsentPageObject.js:42]: Actually, both of these `get*ButtonLabel` methods already exist on BaseFormObject and on BasePageObject there is already `getSaveButtonLabel`

File: test/testcafe/spec/EnduserEmailConsentView_spec.js [hunk 2]
```diff
@@ -34,13 +34,13 @@ async function setup(t) {
 test
   .requestHooks(enduserEmailConsentSuccess)('has the right title, info and button texts', async t => {
     const consentPage = await setup(t);
-    await checkA11y(t);
```
Comment (shawyu-okta) [inline test/testcafe/spec/EnduserEmailConsentView_spec.js:37]: Was removing the `checkA11y` intentional?

Comment (vijetmahabaleshwar-okta) [reply test/testcafe/spec/EnduserEmailConsentView_spec.js:37]: Not intentional. 😬  Good catch!

File: test/testcafe/spec/EnduserEmailConsentView_spec.js [hunk 3]
```diff
@@ -69,14 +70,13 @@ test
...
-    await t.expect(terminalPage.getMessages()).contains('To continue, please return to the original browser tab or window you used to verify.');
-    await t.expect(terminalPage.getMessages()).contains('Close this window anytime.');
+    await t.expect(terminalPage.getMessages(0)).contains('To continue, please return to the original browser tab or window you used to verify.');
+    await t.expect(terminalPage.getMessages(1)).contains('Close this window anytime.');
   });
 
 test
   .requestHooks(requestLogger, enduserEmailConsentFailure)('consent denied flow', async t => {
     const consentPage  = await setup(t);
-    await checkA11y(t);
```
Comment (glenfannin-okta) [inline test/testcafe/spec/EnduserEmailConsentView_spec.js:79]: Did you mean to remove this?

File: test/testcafe/spec/EnduserEmailConsentView_spec.js [hunk 4]
```diff
@@ -50,7 +52,6 @@ test
 test
   .requestHooks(requestLogger, enduserEmailConsentSuccess)('consent granted flow', async t => {
     const consentPage  = await setup(t);
-    await checkA11y(t);
```
Comment (glenfannin-okta) [inline test/testcafe/spec/EnduserEmailConsentView_spec.js:53]: Did you mean to remove this?

**Review by glenfannin-okta — APPROVED** [review]

File: test/testcafe/framework/page-objects/EnduserConsentPageObject.js (hunk 1 above)
Comment (shawyu-okta) [reply test/testcafe/framework/page-objects/EnduserConsentPageObject.js:35]: Can you update to use `this.form.getCancelButtonLabel()`?

**Review by shawyu-okta — APPROVED** [review]
Just the one outstanding comment

**Pull request merged**